router/usecase/router_usecase.go

```diff
@@ -448,6 +453,27 @@ func (r *routerUseCaseImpl) GetCustomDirectQuote(ctx context.Context, tokenIn sd
 	return getBestSingleRouteQuote(ctx, tokenIn, routes, r.logger)
 }
 
+// GetCustomDirectQuoteMultiPool implements mvc.RouterUsecase.
+func (r *routerUseCaseImpl) GetCustomDirectQuoteMultiPool(ctx context.Context, tokenIn sdk.Coin, tokenOutDenom string, poolID []uint64) (domain.Quote, error) {
+	var result quoteImpl
+	for _, v := range poolID {
+		quote, err := r.GetCustomDirectQuote(ctx, tokenIn, tokenOutDenom, v)
+		if err != nil {
+			return nil, err
+		}
+
+		i, ok := quote.(*quoteImpl)
+		if !ok {
+			return nil, fmt.Errorf("quote ( v=%v ) is not *quoteImpl", quote)
+		}
```

Comment:
nit: I don't have any specific recommendations but could you check if there is a way we could avoid casting?

If non-trivial, feel free to resolve

Reply:
I have updated as per your suggestion.